### PR TITLE
Add generic tests and remove borrowed/owned tests

### PR DESCRIPTION
**Pull request by desaikd**
*Issue #232*

*Description of changes:*
This PR works on creating more tests for generic test suite and remove the owned/borrowed tests entirely.

*Changes:*
- Added generic tests for all Ion types
- Added `with_annotations` for adding annotations to `Element`.
- remove owned/borrowed tests

### CODE REVIEW

**Review by almann — COMMENTED** [review]
Looks pretty good--my real only concern is making the `with_annotations` generic.

File: src/value/mod.rs [hunk 1]
```diff
@@ -878,64 +879,106 @@ mod generic_value_tests {
         }
     }
 
-    fn bool_case<E: Element>() -> Case<E> {
+    fn bool_case<E: Element>() -> Case<E>
+    where
+        E: Debug + PartialEq,
```
Comment (almann) [inline src/value/mod.rs:884]: Should we just add this super-trait on `Element` and friends?

**Review by almann — APPROVED** [review]
LGTM, I assume the coverage looks reasonable?

File: src/value/mod.rs (hunk 1 above)
Comment (desaikd) [reply src/value/mod.rs:884]: Yes. I will add it.

**Pull request merged**